code/modules/projectiles/projectile/magic.dm

```diff
@@ -34,6 +34,7 @@
 			return
 		if(target.anti_magic_check())
 			target.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			return
```

Comment:
it didn't have a return so it output the anti magic message but still went through with the effects